controllers/provisioning_controller.go

```diff
@@ -54,7 +54,7 @@ type ProvisioningReconciler struct {
 	Log            logr.Logger
 	OSClient       osclientset.Interface
 	EventRecorder  record.EventRecorder
-	kubeClient     kubernetes.Interface
+	KubeClient     kubernetes.Interface
```

Comment:
What is the difference between this client and the one on line 52?

Reply:
KubeClient always writes and reads from the API server but the Client reads from a cache but writes to the API server. While creating the Secrets, I think either clients would be OK to do the job. But when we need to read the Secrets to eventually create the Deployment object, using a client that did not read from the local cache seemed like a more predictable approach to take.

Reply:
Interesting. Is that what other operators do?

Reply:
Yes, other operators directly use the K8s client. The "Client" was added/generated by KubeBuilder and it *might* work even with the caching. The current focus is on getting metal3 deployment working using the k8s client. Once we have that, we could experiment using this client over the k8s client.